packages/gatsby-plugin-functions/src/gatsby-node.ts

```diff
@@ -0,0 +1,88 @@
+import fs from "fs-extra"
+import glob from "globby"
+import path from "path"
+import webpack from "webpack"
+
+import { ParentSpanPluginArgs, PluginOptions } from "gatsby"
+
+export async function onPreInit(
+  { reporter }: ParentSpanPluginArgs,
+  { path: functionsDirectoryPath }: PluginOptions
+): Promise<void> {
+  const activity = reporter.activityTimer(`building functions`)
+  activity.start()
+
+  const functionsDirectory = path.resolve(
+    process.cwd(),
+    functionsDirectoryPath as string
+  )
+
+  const functionsGlob = `**/*.{js,ts}`
```

Comment:
we should use `resolvableExtensions` instead of hard-coding this — that way I can write my functions in coffeescript 😇 

https://github.com/gatsbyjs/gatsby/blob/2b494c34b5a8bb798fa1134a8a482165afc72fce/packages/gatsby/src/services/initialize.ts#L504-L516

Reply:
Any transpile-to-js format should be supported.

Reply:
Good idea!